istioctl/cmd/revision.go

```diff
@@ -0,0 +1,1079 @@
+		if iop == nil {
+			continue
+		}
+		rev := renderWithDefault(iop.Spec.GetRevision(), "default")
+		ri, revPresent := revisions[rev]
+		if revPresent {
+			iopInfo := IstioOperatorCRInfo{
+				Namespace:      iop.GetNamespace(),
+				Name:           iop.GetName(),
+				Profile:        iop.Spec.GetProfile(),
+				Components:     getEnabledComponents(iop.Spec),
+				Customizations: nil,
+			}
+			if args.verbose {
+				iopInfo.Customizations, err = getDiffs(iop, revArgs.manifestsPath, iop.Spec.GetProfile(), logger)
+				if err != nil {
+					return fmt.Errorf("error while finding customizations: %v", err)
+				}
+			}
+			ri.IstioOperatorCRs = append(ri.IstioOperatorCRs, iopInfo)
+		}
+	}
+
+	if args.verbose {
+		for rev, desc := range revisions {
+			revClient, err := newKubeClientWithRevision(kubeconfig, configContext, rev)
+			if err != nil {
+				return fmt.Errorf("failed to get revision based kubeclient for revision: %s", rev)
+			}
+			if err = enrichWithControlPlanePodInfo(desc, revClient); err != nil {
+				return fmt.Errorf("failed to get control plane pods for revision: %s", rev)
+			}
+			if err = enrichWithGatewayInfo(desc, revClient); err != nil {
+				return fmt.Errorf("failed to get gateway pods for revision: %s", rev)
+			}
+		}
+	}
+
+	switch revArgs.output {
+	case jsonFormat:
+		return printJSON(writer, revisions)
+	default:
+		return printRevisionInfoTable(writer, args.verbose, revisions)
+	}
+}
+
+func printRevisionInfoTable(writer io.Writer, verbose bool, revisions map[string]*RevisionDescription) error {
+	if err := printSummaryTable(writer, verbose, revisions); err != nil {
+		return fmt.Errorf("failed to print summary table: %v", err)
+	}
+	if verbose {
+		if err := printControlPlaneSummaryTable(writer, revisions); err != nil {
+			return fmt.Errorf("failed to print control plane table: %v", err)
+		}
+		if err := printGatewaySummaryTable(writer, revisions); err != nil {
+			return fmt.Errorf("failed to print gateway summary table: %v", err)
+		}
+	}
+	return nil
+}
+
+func printControlPlaneSummaryTable(w io.Writer, revisions map[string]*RevisionDescription) error {
+	fmt.Fprintf(w, "\nCONTROL PLANE:\n")
+	tw := new(tabwriter.Writer).Init(w, 0, 8, 1, ' ', 0)
+	fmt.Fprintf(tw, "REVISION\tISTIOD ENABLED\tCONTROL-PLANE PODS\n")
+	for rev, rd := range revisions {
+		isIstiodEnabled := false
+	outer:
+		for _, iop := range rd.IstioOperatorCRs {
+			for _, c := range iop.Components {
+				if c == "istiod" {
+					isIstiodEnabled = true
+					break outer
+				}
+			}
+		}
+		maxRows := max(1, len(rd.ControlPlanePods))
+		for i := 0; i < maxRows; i++ {
+			var rowRev, rowEnabled, rowPod string
+			if i == 0 {
+				rowRev = rev
+				if isIstiodEnabled {
+					rowEnabled = "YES"
+				} else {
+					rowEnabled = "NO"
+				}
+			}
+			if i < len(rd.ControlPlanePods) {
+				rowPod = fmt.Sprintf("%s/%s",
+					rd.ControlPlanePods[i].Namespace,
+					rd.ControlPlanePods[i].Name)
+			} else if i == 0 && len(rd.ControlPlanePods) == 0 {
```

Comment:
more idiomatic to use switch/case rather than if/else blocks. up to you.

Reply:
Ah! That's neat :) Changed it.